### PR TITLE
[Improve] mysql, postgresql, h2 menu data keep consistent

### LINKED ISSUE #2595: [Improve] mysql, postgresql, h2 menu data keep consistent
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

mysql, postgresql, h2 menu data keep consistent

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by muchunjin**
## What changes were proposed in this pull request

Issue Number: close #2595  

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by wolfboys** [comment]
cc @1996fanrui PTAL, thx

**Comment by RocMarshal** [comment]
Thanks for @macksonmu your contribution. 
I'm confused why the corresponding change of upgrade sql file of postgresql  not exist ? postgre sql was supported since 2.0.
As you know, if the sql file of postgresql was lacked, the users who deployed the database service in postgre would not get the latest version release. 
Please correct me if I'm wrong.
CC @1996fanrui @wolfboys

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 1]
```diff
@@ -88,52 +42,184 @@ create table `t_yarn_queue` (
...
+insert into `t_menu` values (130501, 130500, 'link view', null, null, 'externalLink:view', null, '1', 1, null, now(), now());
+insert into `t_menu` values (130502, 130500, 'link create', null, null, 'externalLink:create', null, '1', 1, null, now(), now());
+insert into `t_menu` values (130503, 130500, 'link update', null, null, 'externalLink:update', null, '1', 1, null, now(), now());
+insert into `t_menu` values (130504, 130500, 'link delete', null, null, 'externalLink:delete', null, '1', 1, null, now(), now());
+insert into `t_menu` values (130601, 130600, 'add yarn queue', null, null, 'yarnQueue:create', '', '1', 0, null, now(), now());
+insert into `t_menu` values (130602, 130600, 'edit yarn queue', null, null, 'yarnQueue:update', '', '1', 0, null, now(), now());
+insert into `t_menu` values (130603, 130600, 'delete yarn queue', null, null, 'yarnQueue:delete', '', '1', 0, null, now(), now());
+
+-- role menu script
+delete from t_role_menu;
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: Would here be compatibility issues if user created some specified roles?
We deleted all records but not fill back the corresponding records based on the new menu encode strategy.

**Review by RocMarshal — COMMENTED** [review]
Thx @macksonmu for the PR.
I checked the file change , the h2 schema & data sql on my PC and the complete & delta upgrade mode of mysql. It ran successfully.

However, back-forward compatibility for users & history records of system is special case we have to consider. So I left a few of comments based on it.

PTAL in your free time. thx~

**Comment by muchunjin** [comment]
> Thanks for @macksonmu your contribution. I'm confused why the corresponding change of upgrade sql file of postgresql not exist ? postgre sql was supported since 2.0. As you know, if the sql file of postgresql was lacked, the users who deployed the database service in postgre would not get the latest version release. Please correct me if I'm wrong. CC @1996fanrui @wolfboys

Thank you so much for your review, I did miss this, I will make it up

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql (hunk 1 above)
Comment (muchunjin) [reply streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: Thanks, I will consider this piece

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 2]
```diff
@@ -88,52 +42,192 @@ create table `t_yarn_queue` (
...
+insert into `t_menu` values (110601, 110600, 'add', null, null, 'member:add', null, '1', 1, null, now(), now());
+insert into `t_menu` values (110602, 110600, 'update', null, null, 'member:update', null, '1', 1, null, now(), now());
+insert into `t_menu` values (110603, 110600, 'delete', null, null, 'member:delete', null, '1', 1, null, now(), now());
+insert into `t_menu` values (110604, 110600, 'role view', null, null, 'role:view', null, '1', 1, null, now(), now());
+insert into `t_menu` values (110605, 110600, 'view', null, null, 'member:view', null, '1', 1, null, now(), now());
+insert into `t_menu` values (120101, 120100, 'add', '/flink/project/add', 'flink/project/Add', 'project:create', '', '0', 0, null, now(), now());
+insert into `t_menu` values (120102, 120100, 'build', null, null, 'project:build', null, '1', 1, null, now(), now());
+insert into `t_menu` values (120103, 120100, 'delete', null, null, 'project:delete', null, '1', 1, null, now(), now());
+insert into `t_menu` values (120104, 120100, 'edit', '/flink/project/edit', 'flink/project/Edit', 'project:update', null, '0', 0, null, now(), now());
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql:100]: AnchorA(a)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 3]
```diff
@@ -88,52 +42,192 @@ create table `t_yarn_queue` (
...
+update `t_role_menu` set menu_id=110602 where menu_id=100049;
+update `t_role_menu` set menu_id=110603 where menu_id=100050;
+update `t_role_menu` set menu_id=110604 where menu_id=100051;
+update `t_role_menu` set menu_id=110605 where menu_id=100063;
+update `t_role_menu` set menu_id=120000 where menu_id=100013;
+update `t_role_menu` set menu_id=120100 where menu_id=100014;
+update `t_role_menu` set menu_id=120101 where menu_id=100017;
+update `t_role_menu` set menu_id=120102 where menu_id=100022;
+update `t_role_menu` set menu_id=120103 where menu_id=100023;
+update `t_role_menu` set menu_id=120104 where menu_id=100035;
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql:174]: AnchorA(b)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 4]
```diff
@@ -88,52 +42,192 @@ create table `t_yarn_queue` (
...
+update `t_role_menu` set menu_id=120306 where menu_id=100067;
+update `t_role_menu` set menu_id=120307 where menu_id=100069;
+update `t_role_menu` set menu_id=130101 where menu_id=100068;
+update `t_role_menu` set menu_id=130102 where menu_id=100034;
+update `t_role_menu` set menu_id=120206 where menu_id=100025;
+update `t_role_menu` set menu_id=130000 where menu_id=100033;
+update `t_role_menu` set menu_id=130401 where menu_id=100040;
+update `t_role_menu` set menu_id=130402 where menu_id=100041;
+
+insert into `t_role_menu` (role_id, menu_id) values (100001, 120104);
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql:237]: AnchorA(c).
(I don't checked it on the PC env. Because I saw the new commits till the current review.)

Will here(all tags bout AnchorA) exist redundant or conflict logic cases ?
- redundant: 
  - There would be no records about menu_id `120104` at `AnchorA(b)`, so the line(`AnchorA(c)`) could be executed.
- conflict:
  - If there are records about (100001, 100035) before executing `AnchorA(b)`, the line `AnchorA(c)`  would fail due to the duplicated unique-key constraint.     

That the case would be `redundant ` or `conflict` depends on the users' status of the production env. It's complex for us to cover.   
Base on mentioned above,      
what is the bounds for the PR back-forward compatibility is impact for the expected state of the delta sql file(2.1.0.sql). 
In other words, should the scope of versions evolution we should do BF-compatibility be `release-2.0.0 -> now` or `release-2.0.0-> some users' cherrypick(into Prod env)  -> now` ?
What about defining the bounds before the next update ?

Looking forward to discussion & suggestions about it.
Thank you~.

**Review by RocMarshal — COMMENTED** [review]
@macksonmu Thank you so much for supplementing the Postgre & the update.
It looks better now. 👍
As you know, we have to be exact due to the next coming release. I left a few of minor comments with my limited read.  It would be good to me after confirming its.
Thank you very much for your patience and excellent effort.

**Review by wolfboys — APPROVED** [review]
LGTM, thanks for your contribution

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql (hunk 4 above)
Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql:237]: > AnchorA(c). (I don't checked it on the PC env. Because I saw the new commits till the current review.)
> 
> Will here(all tags bout AnchorA) exist redundant or conflict logic cases ?
> 
> * redundant:
>   
>   * There would be no records about menu_id `120104` at `AnchorA(b)`, so the line(`AnchorA(c)`) could be executed.
> * conflict:
>   
>   * If there are records about (100001, 100035) before executing `AnchorA(b)`, the line `AnchorA(c)`  would fail due to the duplicated unique-key constraint.
> 
> That the case would be `redundant ` or `conflict` depends on the users' status of the production env. It's complex for us to cover. Base on mentioned above, what is the bounds for the PR back-forward compatibility is impact for the expected state of the delta sql file(2.1.0.sql). In other words, should the scope of versions evolution we should do BF-compatibility be `release-2.0.0 -> now` or `release-2.0.0-> some users' cherrypick(into Prod env) -> now` ? What about defining the bounds before the next update ?
> 
> Looking forward to discussion & suggestions about it. Thank you~.

Thanks for your efforts for current review
